### DISCUSSION TITLE
How do I use middleware with best practices.

### DISCUSSION THREAD

**Discussion by oceanm7n** [eternnoir/pyTelegramBotAPI#1685] [General]
I want to use a middleware to incorporate custom service logic into my bot, that requires it to stop processing a command if certain conditions are met.

How do I explicitly stop every message_handler decorator from processing a command if various conditions are met inside the middleware?

**Comment by coder2020official** [comment]
You need to return either [telebot.handler_backends.CancelUpdate](https://pytba.readthedocs.io/en/latest/sync_version/index.html#telebot.handler_backends.CancelUpdate) or [telebot.handler_backends.SkipHandler](https://pytba.readthedocs.io/en/latest/sync_version/index.html#telebot.handler_backends.SkipHandler)

Returning SkipHandler will skip the handler execution and will call your post_process middleware function.
Returning CancelUpdate will cancel execution of both post_process function, and your handler.

Here's the example of using CancelUpdate:
![image](https://user-images.githubusercontent.com/63740404/186416911-50442d31-ee27-438a-a4c9-f7eeb5c8cc41.png)
[Link](https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/middleware/class_based/antiflood_middleware.py)
